model/testing/src/icon4py/model/testing/helpers.py

```diff
@@ -199,8 +211,7 @@ def __init_subclass__(cls, **kwargs):
         # reflect the name of the test we do this dynamically here instead of using regular
         # inheritance.
         super().__init_subclass__(**kwargs)
-        setattr(cls, f"test_{cls.__name__}", _test_validation)
-        setattr(cls, f"test_{cls.__name__}_benchmark", _test_execution_benchmark)
+        setattr(cls, f"test_{cls.__name__}", _test_and_benchmark)
 
 
 def reshape(arr: np.ndarray, shape: tuple[int, ...]):
```

Comment:
Not related to your PR, but since you're touching already this file, maybe you could the proper return type hint here.